ssawg_trending_scraper.py

```diff
@@ -43,112 +40,59 @@
-                    page_request = requests.get(url)
-                if page_request.status_code == 200:
-                    # use astropy Table 
-                    table_page = Table.read(page_request.text, format='ascii.html', htmldict={'table_id': 2})
-                    # pull quarterly start and stop dates from page
-                    start_time = table_page['TSTART'][0] 
-                    stop_time = table_page['TSTOP'][0]
-                    stop_minus_start = Chandra.Time.date2secs(stop_time) - Chandra.Time.date2secs(start_time)
-                    # define halfway through the quarter
-                    halfway = Chandra.Time.date2secs(start_time) + stop_minus_start
-                    # is now > 50% through quarter?
-                    if Chandra.Time.date2secs(datetime.now()) > halfway: 
-                        return f'{URL_ASPECT}/{self.page}/{year}/Q{quarter}/'
-                    # if not 50% through and it's the first quarter of the year
-                    elif quarter == 1:
-                        # switch to fourth quarter of previous year
-                        return f'{URL_ASPECT}/{self.page}/{year-1}/Q4/' 
-                    else:
-                        # try previous quarter
-                        return f'{URL_ASPECT}/{self.page}/{year}/Q{quarter-1}/' 
-                else:
-                    continue
-
-        def get_proper_urls_perigee(page):
-            """
-            Get the correct URL for the monthly perigee page; 50% through month.
-            """
-            now = datetime.now()
-            # if ~halfway through the month
-            if now.day > 15:
-                return f'{URL_ASPECT}/{self.page}/SUMMARY_DATA/{now.year}-M{now.month:02}/'
-            else:
-                last_month = datetime.now() + timedelta(days=-27)
-                return f'{URL_ASPECT}/{self.page}/SUMMARY_DATA/{last_month.year}-M{last_month.month:02}/'
-
-        # Generate the correct URLs for trending
-        if self.page in trending_pages["Reports"]:
-            self.url = get_proper_urls_reports(self.page)
-        elif self.page in trending_pages["Perigee"]:
-            self.url = get_proper_urls_perigee(self.page)
-        elif self.page in trending_pages["Generic"]:
-            self.url = f'{URL_ASPECT}/{page}/'
+        self.url = self.get_url()
 
         # Generate the page requests and verify page is accessible
-        if self.page == "periscope_drift_reports":
-            self.page_request = requests.get(self.url, auth=(USER_PERISCOPE, PASSWORD))
-        else:
-            self.page_request = requests.get(self.url)
-        if self.page_request.status_code == 200:
-            pass
-        else:
-            raise RuntimeError(f'Investigate issues with {self.page}')
+        self.page_request = self.get_page_request()
 
         self.url_text = self.page_request.text
         self.soup = BeautifulSoup(self.url_text, "lxml")
 
-        def get_elements(soup, element):
```

Comment:
I probably told you to make these into methods of the class, but putting them inside `__init__` was not the intent.  At the end I think these look more like static helper functions that belong outside the class: they are doing very generic things that don't specifically relate to the class object.  But that could go either way.